### PR TITLE
Make dsync as generic distributed sync.mutex

### PR DESCRIPTION
**Pull request by balamurugana**
Previously dsync was using more specific to minio server code base.
This patch makes dsync as generic distrbuted sync.mutex package.

### CODE REVIEW

File: chaos/net-rpc-client.go [hunk 1]
```diff
@@ -75,10 +76,7 @@ func (rpcClient *ReconnectRPCClient) dialRPCClient() (*rpc.Client, error) {
 }
 
 // Call makes a RPC call to the remote endpoint using the default codec, namely encoding/gob.
-func (rpcClient *ReconnectRPCClient) Call(serviceMethod string, args interface {
```
Comment (harshavardhana) [inline chaos/net-rpc-client.go]: With this removal where does the Token and Timestamp will be validated? .  Is the purpose to remove that to not enforce authentication behavior?

Comment (balamurugana) [inline chaos/net-rpc-client.go]: dsync does not know/worry about how server/client is authenticated.   It facilitates networked sync.mutex.

In Minio server use case, minio will do authentication based on JWT+timestamp in `lock-client` to `lock-server` where lock-client is interface compatible to `dsync.NetLocker`.

Let me add a token based Login/Logout method in chaos test for sake of understanding.

Comment (harshavardhana) [inline chaos/net-rpc-client.go]: > Let me add a token based Login/Logout method in chaos test for sake of understanding.

Write it outside of chaos in a separate directory as an example which talks about authenticated lock client.

Comment (balamurugana) [inline chaos/net-rpc-client.go]: Done

**Pull request merged**